### DISCUSSION TITLE
Entity-Attribute-Value paradigm for relational databases

### DISCUSSION THREAD

**Discussion by Dariusms** [linkml/linkml#1923] [Q&A]
How does one model the entity-attribute-value paradigm for relational databases within LinkML. I have tried using multivalued: true, and that does indeed create a relationship table that covers the Entity-Attribute part of it, but I do not see a way of also including a value.

**Comment by cmungall** [comment]
There are a few ways to do this:

1. A single EAV class with 3 slots, and then include this in a collection/container
2. An E class and an AV class, and a multivalued slot such that one E can have many AVs

Of course there are many other considerations: is the `v` typed to Any (in which case, how do you map to a RDBMS if required?) Do you have `v_string`, `v_int`, ...?

**Reply by Dariusms** [reply]
Thank you, method 1 worked. I was thinking varchar2, so strings.
